Scripts/datahandling/resultdata.py

```diff
@@ -115,7 +115,9 @@ def print_matrices(self,
                     sheet_name=f"{description}_{key}")
         # Create text file
         stacked_matrices = pandas.concat(
-            {f"{description}_{key}": df.stack() for key, df in data.items()},
-            names=["purpose", "orig", "dest"])
+            {key: df.stack() for key, df in data.items()},
+            names=["mode", "orig", "dest"])
         stacked_matrices.name = "nr_tours"
-        self.print_concat(stacked_matrices, filename + ".txt")
+        stacked_matrices = stacked_matrices.reset_index()
+        stacked_matrices.insert(0, 'purpose', description)
+        self.print_concat(stacked_matrices.set_index("purpose"), filename + ".txt")
```

Comment:
These additions are then unnecessary

```suggestion
        self.print_concat(stacked_matrices, filename + ".txt")
```